ironfish/src/blockchain/blockchain.ts

```diff
@@ -347,18 +351,18 @@ export class Blockchain {
           throw new VerifyError(VerificationResultReason.ORPHAN)
         }
 
-        await this.connect(block, previous, tx)
+        isFork = (await this.connect(block, previous, tx)).isFork
 
         await this.resolveOrphans(block)
       })
     } catch (e) {
       if (e instanceof VerifyError) {
-        return { isAdded: false, reason: e.reason, score: e.score }
+        return { isAdded: false, isFork: null, reason: e.reason, score: e.score }
       }
       throw e
     }
 
-    return { isAdded: true, reason: null, score: null }
+    return { isAdded: true, isFork: isFork, reason: null, score: null }
```

Comment:
```suggestion
    const isFork = !this.head.hash.equals(block.header)
    return { isAdded: true, isFork: isFork, reason: null, score: null }
```

Reply:
sounds good. Why do we need a mutex lock here? Or did you just make a typo?

Reply:
Two add blocks running at the same time can cause bugs.